drivers/serial/uart_mspm0g3xxx.c

```diff
@@ -186,20 +180,16 @@ static void uart_mspm0g3xxx_isr(const struct device *dev)
 	const struct uart_mspm0g3xxx_config *config = dev->config;
 	struct uart_mspm0g3xxx_data *const dev_data = dev->data;
 
-	/* Get the pending interrupt */
-	int int_status = DL_UART_getEnabledInterruptStatus(
-		config->regs, UART_MSPM0_RX_INTERRUPTS | UART_MSPM0_TX_INTERRUPTS);
+	/* pending interrupt will be used by the provided callback */
+	dev_data->pending_interrupt = DL_UART_getPendingInterrupt(config->regs);
 
 	/* Perform callback if defined */
 	if (dev_data->cb) {
 		dev_data->cb(dev, dev_data->cb_data);
 	}
 
-	/*
-	 * Clear interrupts only after cb called, as Zephyr UART clients expect
-	 * to check interrupt status during the callback.
-	 */
-	DL_UART_clearInterruptStatus(config->regs, int_status);
+	/* clear in case someone accidentally calls the irq methods outside of the ISR */
+	dev_data->pending_interrupt = DL_UART_IIDX_NO_INTERRUPT;
```

Comment:
I don't need this specifically, so it's fine to not include it :) Just wanted to check about the state of the system and the calls etc :)

Reply:
Alright, I'll remove it and squash the other fixup :+1: